tests/approximate-qvm-tests.lisp

```diff
@@ -0,0 +1,78 @@
+(in-package #:qvm-tests)
+
+
+(deftest test--approx-qvm ()
+  ;; Test that the correct kraus operators are set for the correct
+  ;; qubits when building an APPROX-QVM
+  (let* ((approx-qvm (make-instance 'qvm::approx-qvm :number-of-qubits 2 :avg-gate-time 1))
+        (numshots 100)
+        (program "DECLARE R0 BIT; X 0; CNOT 0 1;  MEASURE 0 R0")
+        (parsed-program (quil:parse-quil program)))
+    (load-program approx-qvm parsed-program :supersede-memory-subsystem t)
+    (setf (qvm::qubit-t1 approx-qvm 0) 5
+          (qvm::qubit-t2 approx-qvm 0) 4
+          (qvm::qubit-t1 approx-qvm 1) 2
+          (qvm::qubit-t2 approx-qvm 1) 3
+          (qvm::avg-gate-time approx-qvm) 1)
+   
+    (let ((t1-q0 (qvm::generate-damping-kraus-ops 5 1))
+          (t1-q1 (qvm::generate-damping-kraus-ops 2 1))
+          (t2-q0 (qvm::generate-damping-dephasing-kraus-ops 4 1))
+          (t2-q1 (qvm::generate-damping-dephasing-kraus-ops 3 1)))
+     
+      (is (eql 1 (qvm::avg-gate-time approx-qvm)))
+      (is (equalp t1-q0 (gethash 0 (qvm::t1-ops approx-qvm))))
+      (is (equalp t1-q1 (gethash 1 (qvm::t1-ops approx-qvm))))
+      (is (equalp t2-q0 (gethash 0 (qvm::t2-ops approx-qvm))))
+      (is (equalp t2-q1 (gethash 1 (qvm::t2-ops approx-qvm)))))))
```

Comment:
In this case the elems of the matrices should be exactly identical bc the same function is being used to generate them.

Reply:
That makes sense. I still feel like `equalp` is depending on internal details of the `magicl:matrix` representation though, and that an explicit elementwise test that only depends on the exported magicl interface would be better.  For instance, I think these `equalp` tests will likely break after Cole's "Magicl NG" PR gets merged, since `magicl:matrix` is switching from a `defstruct` to a `defclass`. That's my $0.02 anyway.